pkg/auth/oidc.go

```diff
@@ -56,8 +55,9 @@ type OIDCAuthenticator struct {
 
 // NewOIDC returns a new OIDC authenticator
 func NewOIDC(config *OIDCAuthConfig) (*OIDCAuthenticator, error) {
-	ctx, cancel := grpcutil.WithDefaultTimeout(context.Background())
-	defer cancel()
+  // Reverting the defaultTimeout base context as some of the coreos oidc api
+  // takes this context for subsequent call and end up using expired context.
+  ctx := context.Background()
```

Comment:
nit: code block does not have the right tab width
```suggestion
    // Reverting the defaultTimeout base context as some of the coreos oidc api
    // takes this context for subsequent call and end up using expired context.
    ctx := context.Background()
```

Reply:
Fixed it.